### DISCUSSION TITLE
Whats necessary to support iOS 14 using Analytics and IDFA?

### DISCUSSION THREAD

**Discussion by ghost** [invertase/react-native-firebase#4282] [Q&A] [answered]
I'm working with several apps that are already in production, all of them use React Native Firebase for analytics and the IDFA, I'm trying to to keep everything OK for iOS 14, but, as you can see in the following image, Google released a SDK version 4 days ago with AppTrackingTransparency stuff .

![image](https://user-images.githubusercontent.com/47385105/93633905-189cee00-f9c6-11ea-859d-2580f013d912.png)

My question is, what do I need to do to keep everything ok with iOS 14? Is there any place with more details about this subject?  Looking at React Native Firebase source code it seems that the latest version of the lib isn't pointing to the latest version of FirebaseSDK, is that right?

Thank you in advance.

**Comment by mikehardy** [comment, marked as answer]
https://github.com/invertase/react-native-firebase/issues/4107

> With the addition of that PR to react-native-permissions to handle the user consent, and the ability (proven) to use the latest version of the AdMobs SDK, I don't think there is anything more to do here. I'm going to close this but if I am wrong obviously we can reopen and discuss - not closing because it's not important just that I think it's resolved :-). Cheers

**Reply by ghost** [reply]
Just to make myself clear. Do I need to do those upgrades if I'm only using Analytics and the device identification as mentioned [here](https://rnfirebase.io/analytics/usage/installation/ios#device-identification) in the docs?

**Reply by mikehardy** [reply]
I believe it will be necessary to request AppTrackingTransparency permission if you touch the IDFA. But Apple App Review process is the law and I will not attempt to guess their disposition. You will likely find more authoritative discussion in the firebase-ios-sdk repo, we are just an open source wrapper for Google's library, and they're the ones that will know more.
